### PR TITLE
[deployment] Move list of services that use platform scaffolding out of deployment-service code into package.meta files.

### LINKED ISSUE #1040: Fix usesPlatformScaffolding
This data should come into the deployment service from some file and not require people to edit deployment service

### PR DESCRIPTION
**Pull request by yzl**
### :nut_and_bolt: Description: What code changed, and why?
Move the list of services that use platform scaffolding out of code and into `package.meta` files.

Fixes #1040.

### :white_check_mark: Checklist

- [ ] Tests added/updated?
- [ ] Docs added/updated?

### :camera: Screenshots, if applicable

### CODE REVIEW

File: components/automate-deployment/pkg/assets/assets.bindata.go [hunk 1]
```diff
@@ -1,19 +1,19 @@
-// Code generated by go-bindata. DO NOT EDIT.
+// Code generated by go-bindata.
```
Comment (jaym) [inline components/automate-deployment/pkg/assets/assets.bindata.go]: is this correct? is it possible your go bindata is old?

File: components/automate-deployment/pkg/server/server.go [hunk 2]
```diff
@@ -437,20 +439,26 @@ func (s *server) configRenderer() (ConfigRenderer, error) {
 			return "", errors.Wrapf(err, "could not converge %s configuration to TOML", service.Name())
 		}
 
-		if usesPlatformScaffolding(service) {
+		pkgsMeta := make([]*product.PackageMetadata, 0, len(s.deployment.ExpectedServices))
```
Comment (jaym) [inline components/automate-deployment/pkg/server/server.go]: why do we need to gather all the metadatas every time for each service? Can we just grab `services.MetadataForPackage(service.Name())`? Or maybe `services.UsesPlatformScaffolding(service.Name())`?

File: components/automate-deployment/pkg/assets/assets.bindata.go (hunk 1 above)
Comment (yzl) [reply components/automate-deployment/pkg/assets/assets.bindata.go]: It is indeed old. Regenerated.

File: components/automate-deployment/pkg/server/server.go [hunk 3]
```diff
@@ -446,14 +447,11 @@ func (s *server) configRenderer() (ConfigRenderer, error) {
...
-	//TODO (jaym): find a better way to get this information
-	switch service.Name() {
-	case "secrets-service", "pg-sidecar-service", "teams-service", "authz-service", "session-service", "automate-dex",
-		"authn-service", "compliance-service", "nodemanager-service", "notifications-service", "applications-service",
-		"automate-cs-bookshelf", "automate-cs-oc-erchef", "automate-cs-oc-bifrost", "ingest-service", "license-control-service":
-		return true
-	}
-	return false
+	metadata := services.MetadataForPackage(service.Name())
+	if metadata == nil || !metadata.UsesPlatformScaffolding {
```
Comment (jaym) [inline components/automate-deployment/pkg/server/server.go]: nit
```
return metadata != nil && metadata.UsesPlatformScaffolding
```

**Pull request merged**